packages/protocol/contracts/dapis/SamplePriceDataFeed.sol

```diff
@@ -2,15 +2,33 @@
 pragma solidity 0.8.4;
 
 import "./RrpDapiServer.sol";
+import "./Api3Adminship.sol";
 
 // A sample price data feed implemented as a dAPI client
-contract SamplePriceDataFeed {
+contract SamplePriceDataFeed is Api3Adminship {
   RrpDapiServer public rrpDapiServer;
+
   bytes16 public latestDapiId;
   int256 public latestAnswer;
 
-  constructor(address _rrpDapiServer) {
+  uint256 public cooldownTime = 1 days; // TODO: is this the right value?
+  mapping(address => uint64) public adminToDapiLastUpdated;
+
+  /// @dev Reverts if the caller tries to update a dAPI again before the cooldown period has elapsed
+  modifier cooldownTimeEnded() {
+    require(
+      adminToDapiLastUpdated[msg.sender] == 0 || // first time updating will not have a last updated value
+        block.timestamp >= (adminToDapiLastUpdated[msg.sender] + cooldownTime),
+      "Cooldown period has not finished"
+    );
+    _;
+  }
+
+  constructor(address _rrpDapiServer, address _metaAdmin) Api3Adminship(_metaAdmin) {
+    require(_rrpDapiServer != address(0), ERROR_ZERO_ADDRESS);
     rrpDapiServer = RrpDapiServer(_rrpDapiServer);
+    // TODO: should we set superadmin status to technical team deploying the contract?
+    // this.setAdminStatus(msg.sender, AdminStatus.SuperAdmin);
   }
 
   function setDapi(
```

Comment:
Do you know when this would be used? I think this is Burak's "reset" function or smth. Can you confirm?

Reply:
Yeah this was a sort of a placeholder